core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -288,7 +297,7 @@ class ReplicaFetcherThread(name: String,
     partitionMap.foreach { case (topicPartition, partitionFetchState) =>
       // We will not include a replica in the fetch request if it should be throttled.
       if (partitionFetchState.isActive && !shouldFollowerThrottle(quota, topicPartition))
-        requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize))
+        requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.fetchOffset, partitionFetchState.logStartOffset, fetchSize))
```

Comment:
`logStartOffset` is also update the of updated FetchRequest and it is needed by the leader to learn the `logStartOffset` of the follower. It seems reasonable to include it in the `FetchRequest.PartitionData` since `FetchRequest.PartitionData` is supposed to contain the same fields of `Protocol.FETCH_REQUEST_PARTITION_V1`. Does this make sense?